src/bundle.rs

```diff
@@ -272,25 +281,65 @@ impl<T: Authorization, V> Bundle<T, V> {
     /// Returns the authorization for this bundle.
     ///
     /// In the case of a `Bundle<Authorized>`, this is the proof and binding signature.
-    pub fn authorization(&self) -> &T {
+    pub fn authorization(&self) -> &A {
         &self.authorization
     }
 
     /// Construct a new bundle by applying a transformation that might fail
     /// to the value balance and balances of assets to burn.
+
+    // FIXME:
+    //
+    // This function is currently used in the `testing` module of the
+    // `librustzcash/zcash_primitives` crate, specifically in the `arb_bundle`
+    // function to convert a ValueSum-based Bundle to an Amount-based Bundle.
+    //
+    // It converts balance and burn values from one type to another (from ValueSum to Amount).
+    // To achieve this, the function uses a complex series of `TryFrom` conversions
+    // and `unwrap` calls. These conversions and unwraps are necessary to convert
+    // value_balance and burn values. The complex construction and additional
+    // `TryFrom` constraints are required because it's the only way to perform
+    // the conversion. This function is restricted to test configuration due
+    // to the use of unwrap calls.
+    //
+    // An alternative approach could be to modify this function to convert only
+    // value_balance without converting burn values, as in practice, we are only
+    // converting types. However, the types for burn values are now always NoteValue.
+    //
+    // WARNING: Contains unwrap() calls! DO NOT remove #[cfg(test)] or handle unwrap() carefully!
+    #[cfg(test)]
```

Comment:
Maybe better to move to test section?

Reply:
Removed `#[cfg(test)]` since `try_map_value_balance` is now simplified. As discussed, we've decided to keep `try_map_value_balance` in its original place.